tests/api_tests/nodes/test_views.py

```diff
@@ -1352,7 +1467,7 @@ def test_filtering_on_invalid_field(self):
         assert_equal(res.status_code, 400)
         errors = res.json['errors']
         assert_equal(len(errors), 1)
-        assert_equal(errors[0]['detail'], 'Query string contains an invalid filter.')
+        assert_equal(errors[0]['detail'], "'invalid' is not a valid field for this endpoint.")
```

Comment:
Without getting too deep into this PR/ conversation, remember that sanitization, validation, and escaping are different things. There may be more than one option here, depending on use case. If query parameters are meant for internal usage (not user-facing display), it may not make sense to sanitize them at all.

We HTML-sanitize values we expect the user to see in their browser when rendered as HTML. If we act on them internally only, then they aren't generally candidates for stored or reflected attacks- and so should not be sanitized. (and bleach wouldn't protect us against nosql injection attacks anyway)